### Task1:
You need to code a program that prints numbers from 1 to 20 except multiple of 2 & 3.

Hint: use "modulo" operator and "and/or" operators

In [6]:
notdiv2_3=[]
for num in range(1,21):
    if num%2!=0 and num%3!=0:
        notdiv2_3.append(num)
notdiv2_3

[1, 5, 7, 11, 13, 17, 19]

### Task2:
Write a program that keep on accepting number from the user until user enters Zero. Display the sum and average of all the numbers.

Hint: Use while loop with flag. 

In [21]:
sum=0
i=0
flag = True
while flag:
    num=input("Enter Numbers or Press Zero to end with sum: ")
    i=i+1
    a=int(num)
    sum=sum+a
    if num=='0':
        flag = False
    else:
        pass
avg=sum/i
print(f"Sum={sum} and Average is {avg}")

Enter Numbers or Press Zero to end with sum: 10
Enter Numbers or Press Zero to end with sum: 20
Enter Numbers or Press Zero to end with sum: 30
Enter Numbers or Press Zero to end with sum: 40
Enter Numbers or Press Zero to end with sum: 0
Sum=100 and Average is 20.0


### Task3:
Accept 10 numbers from the user and display their average.

Hint: Use loop for accepting numbers or use split to take input in one go

In [45]:
i=1
sum=0
while (i<=10):
    num=int(input("Number %d: "%i))
    sum=sum+num
    i=i+1
avg=sum/10
print(f"Average : {avg}")

Number 1: 7
Number 2: 4
Number 3: 5
Number 4: 4
Number 5: 5
Number 6: 6
Number 7: 5
Number 8: 5
Number 9: 65
Number 10: 5
Average : 11.1


In [17]:
sum=0
i=0
while i<10:
    num=input(f"Please Enter Number {i+1}: ")
    a=int(num)
    sum=sum+a
    i=i+1
avg=sum/10
print (avg)


Please Enter Number 1: 5
Please Enter Number 2: 5
Please Enter Number 3: 5
Please Enter Number 4: 5
Please Enter Number 5: 5
Please Enter Number 6: 5
Please Enter Number 7: 5
Please Enter Number 8: 5
Please Enter Number 9: 8
Please Enter Number 10: 5
5.3


### Task4:
Write a program to display sum of odd numbers and even numbers that fall between 12 and 37(including both numbers)


In [14]:
evens=[]
odds=[]
for a in range(11,38):
    if a%2==0:
        evens.append(a)
    else:
        odds.append(a)
print(evens)
print(odds)


[12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34, 36]
[11, 13, 15, 17, 19, 21, 23, 25, 27, 29, 31, 33, 35, 37]


### Task5:
Write a program to print table of a number accepted from user.


In [38]:
table=int(input("Enter a Number(Require Table): "))
for a in range(1,11):
    ans=table*a
    print(f"{table} * {a} = {ans}")
    a=a+1

Enter a Number(Require Table): 7
7 * 1 = 7
7 * 2 = 14
7 * 3 = 21
7 * 4 = 28
7 * 5 = 35
7 * 6 = 42
7 * 7 = 49
7 * 8 = 56
7 * 9 = 63
7 * 10 = 70


### Task6:
Write a program to display product of the digits of a number accepted from the user.

If user inputs:123 Outputs 6

If user inputs: 567 Outputs 210

In [149]:
count=0
e=1
c=0
num=[]
num=input("Enter Numbers to find their product: ")
nums=int(num)
while nums>0:
    nums=nums//10
    count=count+1
    d=int(num[c])
    c=c+1
    f=d*e
    e=f
    a=count
print(e)

Enter Numbers to find their product: 123456
720


### Task7: 
Write a program to print first 10 even numbers in reverse order.

In [84]:
even=[]
i=21
while i>1:
    i -=1
    if i%2==0:
        even.append(i)
even

[20, 18, 16, 14, 12, 10, 8, 6, 4, 2]

### Task8:
Write a program to find the sum of the digits of a number accepted from user

If user inputs:123 Outputs 6

If user inputs: 567 Outputs 18

In [152]:
count=0
e=0
c=0
num=[]
num=input("Enter Numbers to find their sum: ")
nums=int(num)
while nums>0:
    nums=nums//10
    count=count+1
    d=int(num[c])
    c=c+1
    f=d+e
    e=f
    a=count
print(e)

Enter Numbers to find their sum: 567
18


### Task9:
Write a program that accepts a sentence and calculate the number of letters and digits.

Suppose the following input is supplied to the program:

hello world! 123

Then, the output should be:

LETTERS 10

DIGITS 3


In [156]:
digit=letter=0
sent=input("Type sentences: ")
for ch in sent:
   if ch.isdigit():
      digit=digit+1
   elif ch.isalpha():
      letter=letter+1
   else:
      pass
print("Letters:", letter)
print("Digits:", digit)

Type sentences: hello world! 123
Letters: 10
Digits: 3


### Task10: 
You need to trace and make a list of all such numbers from1000 to 3000 in which all the digits are even numbers.

In [85]:
evens=[]
i=1000
while i<=3000:
    i +=1
    if i%2==0:
        evens.append(i)
evens

[1002,
 1004,
 1006,
 1008,
 1010,
 1012,
 1014,
 1016,
 1018,
 1020,
 1022,
 1024,
 1026,
 1028,
 1030,
 1032,
 1034,
 1036,
 1038,
 1040,
 1042,
 1044,
 1046,
 1048,
 1050,
 1052,
 1054,
 1056,
 1058,
 1060,
 1062,
 1064,
 1066,
 1068,
 1070,
 1072,
 1074,
 1076,
 1078,
 1080,
 1082,
 1084,
 1086,
 1088,
 1090,
 1092,
 1094,
 1096,
 1098,
 1100,
 1102,
 1104,
 1106,
 1108,
 1110,
 1112,
 1114,
 1116,
 1118,
 1120,
 1122,
 1124,
 1126,
 1128,
 1130,
 1132,
 1134,
 1136,
 1138,
 1140,
 1142,
 1144,
 1146,
 1148,
 1150,
 1152,
 1154,
 1156,
 1158,
 1160,
 1162,
 1164,
 1166,
 1168,
 1170,
 1172,
 1174,
 1176,
 1178,
 1180,
 1182,
 1184,
 1186,
 1188,
 1190,
 1192,
 1194,
 1196,
 1198,
 1200,
 1202,
 1204,
 1206,
 1208,
 1210,
 1212,
 1214,
 1216,
 1218,
 1220,
 1222,
 1224,
 1226,
 1228,
 1230,
 1232,
 1234,
 1236,
 1238,
 1240,
 1242,
 1244,
 1246,
 1248,
 1250,
 1252,
 1254,
 1256,
 1258,
 1260,
 1262,
 1264,
 1266,
 1268,
 1270,
 1272,
 1274,
 1276,
 1278,
 1280,
 1282,
 1284,
 1286,

### Task11: 
You need to write a program that counts the number of vowels in a given sentence as input from console. 

Suppose the following input is supplied to the program:

Hello world! Practice makes perfect

Then, the output should be:

a appeared 2 times

e appeared 5 times

i appeared  1 time

o appeared 2 times

u appeared 0 time


In [98]:
i=0
sentence = input('Sentence: ').lower()
while i<5:
    letter=0
    vowels=['a','e','i','o','u']
    for ch in sentence:
        if ch.count(vowels[i]):
            letter=letter+1
    if letter<=1:
        t='time'
    else:
        t='times'
    print(f'{vowels[i]} appeared {letter} {t}')
    i=i+1

Sentence: Hello world! Practice makes perfect
a appeared 2 times
e appeared 5 times
i appeared 1 time
o appeared 2 times
u appeared 0 time


### Task12:
You need to write a program that takes sequence of lines as input and prints the lines after making all characters in the sentence capitalized.

Suppose the following input is supplied to the program:

Hello world

Practice makes perfect

Then, the output should be:

HELLO WORLD
PRACTICE MAKES PERFECT

In [169]:
sent=input("Type Sentence: ").upper()
print(sent)

Type Sentence: Hello world  Practice makes perfect
HELLO WORLD  PRACTICE MAKES PERFECT


### Task13:
You need to write a code that takes a comma separated sequence of words as input and prints the words in a comma-separated sequence after sorting them alphabetically.

Suppose the following input is supplied to the program:

without,hello,bag,world

Then, the output should be:

bag,hello,without,world

In [9]:
sent=["without","hello","bag","world"]
sent.sort()
print(sent)

['bag', 'hello', 'without', 'world']


### Task14:
You need code a program that calculates and prints the value according to the given formula:
P = Square root of [(2 * A * B)/C]

Following are the fixed values of A and B:

A is 50. B is 30.

The values of the literal C should be taken as console input to your program in a comma-separated sequence.

Example:
Let us assume the following comma separated input sequence is given to the program:
100,150,180

The output of the program should be:
18,22,24

In [38]:
a=50
b=30
c=int(input("Enter C: "))
p=(2*a*b/c)**0.5
print(p)

Enter C: 180
12.909944487358056


### Task15:
You need to write a ocde in python that finds all such numbers which are divisible by 7 but are not a multiple of 5,
between 2000 and 3200 (both included).

Print the obtained numbers in a comma-separated sequence on a single line

Hint: modulo operator, and/or operator, for loop /while loop

In [41]:
d7_n5=[]
for a in range(1999,3201):
    if a%7==0:
        if a%5!=0:
            d7_n5.append(a)
print(d7_n5)

[2002, 2009, 2016, 2023, 2037, 2044, 2051, 2058, 2072, 2079, 2086, 2093, 2107, 2114, 2121, 2128, 2142, 2149, 2156, 2163, 2177, 2184, 2191, 2198, 2212, 2219, 2226, 2233, 2247, 2254, 2261, 2268, 2282, 2289, 2296, 2303, 2317, 2324, 2331, 2338, 2352, 2359, 2366, 2373, 2387, 2394, 2401, 2408, 2422, 2429, 2436, 2443, 2457, 2464, 2471, 2478, 2492, 2499, 2506, 2513, 2527, 2534, 2541, 2548, 2562, 2569, 2576, 2583, 2597, 2604, 2611, 2618, 2632, 2639, 2646, 2653, 2667, 2674, 2681, 2688, 2702, 2709, 2716, 2723, 2737, 2744, 2751, 2758, 2772, 2779, 2786, 2793, 2807, 2814, 2821, 2828, 2842, 2849, 2856, 2863, 2877, 2884, 2891, 2898, 2912, 2919, 2926, 2933, 2947, 2954, 2961, 2968, 2982, 2989, 2996, 3003, 3017, 3024, 3031, 3038, 3052, 3059, 3066, 3073, 3087, 3094, 3101, 3108, 3122, 3129, 3136, 3143, 3157, 3164, 3171, 3178, 3192, 3199]


### Task16:
With a given integral number n, write a program to generate a dictionary that contains (i, i*i) such that is an integral number between 1 and n (both included). and then the program should print the dictionary.

Suppose the following input is supplied to the program:

8

Then, the output should be:

{1: 1, 2: 4, 3: 9, 4: 16, 5: 25, 6: 36, 7: 49, 8: 64}

In [64]:
dic={}
num=int(input("Square sequance till enter number:"))
for i in range(1,num+1):
    square=i*i
    dic[i]=square
print(dic)

Square sequance till enter number: 8
{1: 1, 2: 4, 3: 9, 4: 16, 5: 25, 6: 36, 7: 49, 8: 64}


### Task17:
You need to write a code which accepts a sequence of comma separated 4 digit binary numbers as its input and then check whether they are divisible by 5 or not. 

The numbers that are divisible by 5 are to be printed in a comma separated sequence.

Example:

0100,0011,1010,1001

Then the output should be:
1010

Reference: 
1) https://www.datacamp.com/.../python-data-type-conversion

2) https://stackoverflow.com/questions/21765779/converting-binary-to-decimal-integer-output?fbclid=IwAR3DD_oiq5pn99IjyxSJgojvB5W6M4JqQtv4EUl_IF_DtmuRXlP-DswcKAA

In [ ]:
# Your code goes here
